addons/account/tests/test_tax_report.py

```diff
@@ -178,3 +178,58 @@ def test_tax_report_change_country(self):
                 # Tags already exist since 'copied_report_1' belongs to 'test_country_2'
                 for tag in line_tags:
                     self.assertIn(tag, country_2_tags_after_change, "The tax_tags expressions sharing their tags with other report should not receive new tags since they already exist.")
+
+    def test_unlink_report_line_tags_archive(self):
+        test_tax = self.env['account.tax'].create({
+            'name': "Test tax",
+            'amount_type': 'percent',
+            'amount': 25,
+            'country_id': self.tax_report_1.country_id.id,
+            'type_tax_use': 'sale',
+            'invoice_repartition_line_ids': [
+                (0, 0, {
+                    'factor_percent': 100,
+                    'repartition_type': 'base',
+                }),
+
+                (0, 0, {
+                    'factor_percent': 100,
+                    'repartition_type': 'tax',
+                    'tag_ids': [(6, 0, self.tax_report_line_1_55.expression_ids._get_matching_tags()[0].ids)],
```

Comment:
Rather than using [0], use a .filtered() call to exclude the - tag (not tag.tax_negate), this way we don't depend on model sequence (in case we change something about it some day).